server/src/test/java/io/crate/types/GeoPointTypeTest.java

```diff
@@ -104,6 +105,14 @@ public void testConversionFromIntegerArray() throws Exception {
         assertThat(value.getY()).isEqualTo(2.0);
     }
 
+    @Test
+    public void test_return_null_when_converting_values_containing_null_to_geo_point() {
+        Point value = DataTypes.GEO_POINT.implicitCast(new Integer[]{1, null});
+        assertThat(value).isNull();
+        value = DataTypes.GEO_POINT.implicitCast(Arrays.asList(null, 1));
+        assertThat(value).isNull();
```

Comment:
I think there are some more cases missing, array of doubles?
Check codecov report complaining for untested lines.